lib/rubocop/cop/style/perl_backrefs.rb

```diff
@@ -5,6 +5,13 @@ module Cop
     module Style
       # This cop looks for uses of Perl-style regexp match
       # backreferences like $1, $2, etc.
+      #
+      # @example
+      #   # bad
+      #   puts $1
+      #
+      #   # good
+      #   puts Regexp.last_match[1]
```

Comment:
Maybe we should prefer using `Regexp.last_match(1)` instead of `Regexp.last_match[1]`. Because the cop auto-corrects bad code to the `()` style. And the `()` style is slightly faster than `[]` style.

```ruby
require 'benchmark'

/a(b)/ =~ 'ab'
Benchmark.bm do |x|
  x.report('()'){50000000.times{Regexp.last_match(1)}}
  x.report('[]'){50000000.times{Regexp.last_match[1]}}
end
```

```
$ ruby bench.rb
       user     system      total        real
()  4.780000   0.000000   4.780000 (  4.782734)
[]  5.470000   0.000000   5.470000 (  5.471679)
```


But we don't have a cop for this style.